src/providers/FusionAuthProvider.tsx

```diff
@@ -1,95 +1,132 @@
             const queryParams = {
-                client_id: clientID,
-                scope: scope,
+                client_id: configuration.clientID,
+                scope: configuration.scope,
                 response_type: 'code',
-                redirect_uri: redirectUri,
+                redirect_uri: configuration.redirectUri,
                 code_challenge: await generatePKCE(),
                 code_challenge_method: 'S256',
                 state: `${generateRandomString()}:${state}`,
             };
             const fullUrl = generateUrl(FunctionType.login, queryParams);
             window.location.assign(fullUrl);
         },
-        [clientID, scope, redirectUri, generateUrl],
+        [configuration, generateUrl],
     );
 
     const logout = useCallback(async () => {
         const queryParams = {
-            client_id: clientID,
-            post_logout_redirect_uri: redirectUri,
-            id_token_hint: idTokenHint ?? '',
+            client_id: configuration.clientID,
+            post_logout_redirect_uri: configuration.redirectUri,
+            id_token_hint: configuration.idTokenHint ?? '',
         };
         const fullUrl = generateUrl(FunctionType.logout, queryParams);
         window.location.assign(fullUrl);
-    }, [clientID, redirectUri, idTokenHint, generateUrl]);
+    }, [configuration, generateUrl]);
 
     const register = useCallback(
         async (state = '') => {
             const queryParams = {
-                client_id: clientID,
-                scope: scope,
+                client_id: configuration.clientID,
+                scope: configuration.scope,
                 response_type: 'code',
-                redirect_uri: redirectUri,
+                redirect_uri: configuration.redirectUri,
                 code_challenge: await generatePKCE(),
                 code_challenge_method: 'S256',
                 state: `${generateRandomString()}:${state}`,
             };
             const fullUrl = generateUrl(FunctionType.register, queryParams);
             window.location.assign(fullUrl);
         },
-        [clientID, scope, redirectUri, generateUrl],
+        [configuration, generateUrl],
     );
 
+    useEffect(() => {
+        try {
+            if (hasAuthParams()) {
+                const urlParams = window.location.search;
+                const queryParams = urlParams.split('&');
+                const parsedQuery: Record<string, any> = {};
+                queryParams.forEach(qp => {
+                    const [key, val] = qp.split('=');
+                    parsedQuery[key] = decodeURIComponent(val);
+                });
+
+                axios
+                    .post(configuration.serverUrl, {
+                        client_id: parsedQuery.client_id,
+                    })
+                    .then(response => {
+                        setUser(response.data.user);
+                    });
+            }
+        } catch (error) {
+            console.log(error);
```

Comment:
Let's use console.error here. We also will probably want a way for users to intercept the error state here so they can handle it in a custom way